src/app/components/client-reviews/client-reviews.component.ts

```diff
@@ -0,0 +1,112 @@
+import { Subscription } from 'rxjs';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ClientReviewsService } from 'src/app/services/client-reviews/client-reviews.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { environment } from 'src/environments/environment';
+import { ToastrService } from 'ngx-toastr';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Reply } from 'src/app/models/client-reviews/review';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-client-reviews',
+  templateUrl: './client-reviews.component.html',
+  styleUrls: ['./client-reviews.component.scss']
+})
+export class ClientReviewsComponent implements OnInit, OnDestroy {
+  subscription: Subscription = new Subscription();
+  count: number;
+  results: any[];
+  url: string;
+  next: string;
+  prev: string;
+  disabledNext = false;
+  disabledPrev = false;
+  profileImage: string;
+  replyUrl: string;
+  replyPost: Reply[];
+  reviewId: number;
+  addReplyToReview: FormGroup;
+
+  constructor(
+    private reviewsService: ClientReviewsService,
+    private route: ActivatedRoute,
+    private toastrService: ToastrService,
+    private router: Router,
+    private spinner: NgxSpinnerService,
+    private formBuilder: FormBuilder
+  ) {
+    this.profileImage = 'assets/img/people.png';
+    this.addReplyToReview = this.formBuilder.group({
+      reply: ['', Validators.compose([Validators.required])]
+    });
+  }
+
+  ngOnInit(): void {
+    this.spinner.show();
+    this.route.paramMap.subscribe(result => {
+      const clientId = result.get('clientId');
+      this.url = `${environment.api_url}/auth/${clientId}/reviews/`;
+      this.fetchReviews(this.url);
+    });
+  }
+
+  fetchReviews(url) {
+    this.subscription.add(
+      this.reviewsService.getReviews(url).subscribe(
+        response => {
+          this.results = response.results;
+          this.count = response.count;
+
+          if (response.next) {
```

Comment:
What do you think of having the `disabledNext` and the `disabledPrev` defaulted to `true`
In that case, when there is `response.next` or `response.previous` we could just set the  values to `false` and not have many `if-else` blocks

```
          if(response.next) {
            this.next = response.next;
            this.disabledNext = false;
          }
          if(response.previous) {
            this.prev = response.previous;
            this.disabledPrev = false;
          }
```

Reply:
Thanks for this @Tevinthuku, Let me reach out to @dorothyas as this code was authored by her and is still under review.